text/0000-rust-governance.md

```diff
@@ -0,0 +1,728 @@
+which should be prioritized.
+
+### Moderation team
+
+Finally, the moderation team is responsible for dealing with CoC violations.
+
+One key difference from the other subteams is that the moderation team does not
+have a leader. Its members are chosen directly by the core team, and should be
+community members who have demonstrated the highest standard of discourse and
+maturity. To limit conflicts of interest, **the moderation subteam should not
+include any core team members**. However, the subteam is free to consult with
+the core team as it deems appropriate.
+
+The moderation team will have a public email address that can be used to raise
+complaints about CoC violations (forwards to all active moderators).
+
+#### Initial plan for moderation
+
+What follows is an initial proposal for the mechanics of moderation. The
+moderation subteam may choose to revise this proposal by drafting an RFC, which
+will be approved by the core team.
+
+Moderation begins whenever a moderator becomes aware of a CoC problem, either
+through a complaint or by observing it directly. In general, the enforcement
+steps are as follows:
+
+> **These steps are adapted from text written by Manish Goregaokar, who helped
+articulate them from experience as a Stack Exchange moderator.**
+
+* Except for extreme cases (see below), try first to address the problem with a
+  light public comment on thread, aimed to de-escalate the situation. These
+  comments should strive for as much empathy as possible. Moderators should
+  emphasize that dissenting opinions are valued, and strive to ensure that the
+  technical points are heard even as they work to cool things down.
+
+  When a discussion has just gotten a bit heated, the comment can just be a
+  reminder to be respectful and that there is rarely a clear "right" answer. In
+  cases that are more clearly over the line into personal attacks, it can
+  directly call out a problematic comment.
+
+* If the problem persists on thread, or if a particular person repeatedly comes
+  close to or steps over the line of a CoC violation, moderators then email the
+  offender privately. The message should include relevant portions of the CoC
+  together with the offending comments. Again, the goal is to de-escalate, and
+  the email should be written in a dispassionate and empathetic way. However,
+  the message should also make clear that continued violations may result in a
+  ban.
+
+* If problems still persist, the moderators can ban the offender. Banning should
+  occur for progressively longer periods, for example starting at 1 day, then 1
+  week, then permanent. The moderation subteam will determine the precise
+  guidelines here.
+
+In general, moderators can and should unilaterally take the first step, but
+steps beyond that (particularly banning) should be done via consensus with the
+other moderators. Permanent bans require core team approval.
```

Comment:
@nrc 

(text re-ordered a bit)

> If part of the reason for having a moderation team is to avoid conflict of interest, it seems you can't have this check - it's clearly a conflict of interest. I realise in purely practical terms this is not a big deal, but appearance is very important, and this creates a bad appearance. In particular, this means that the core team are essentially exempt from being banned and, a cynical person could argue, from the CoC itself.

> We should trust out moderation team by default, if they turn out to be terrible then I believe we have the ability and community will to fix things at that point rather than trying to plan in advance.

I don't believe that conflicts of interest or trust are such binary things. This RFC gives a *huge* amount of trust to subteams, allowing them to accept, reject, and implement RFCs independently, and even to set the policy about what requires an RFC in their area. Likewise, the majority of the moderation subteam's activities are done without oversight, including temporary bans.

Similarly, prohibiting core team members from sitting on the moderation team substantially decreases the potential for conflicts of interest -- the most important of which, historically, is not the claim that the core team is breaking the CoC but rather that any attempts to enforce it are actually about squelching dissent on technical topics.

In short, I object to the idea that having any amount of oversight indicates a default of mistrust, or that any potential for conflicts of interest indicates that moderation is a sham. It is all a spectrum, and I believe that this RFC moves control to a substantial degree *away* from the core team, keeping around only final gatekeeping checks at the final stage. It is for catching mistakes in the process that potentially have significant consequences.

> A ban is not a death penalty - it can always be fixed afterwards so there is plenty of opportunity for the core team to work with the moderation team if a mistake is made and to fix the mistake. It seems overly cautious to have this check - the expense of the paranoia about the moderation team going overboard (which seems unlikely) is the perception of corruption.

A ban might be reversible, but the fallout from making a ban, and then reversing it, could be extreme. Much better to avoid the mistake in the first place.

As I said above, the worry about conflicts of interest that I've encountered has wholly been about the other side -- that if the core team enforced the CoC, it could be seen as doing so to silence technical dissent.